pilot/adapter/config/file/README.md

```diff
@@ -0,0 +1,39 @@
+CRD File Monitor
+======================
+
+This package provides the ability to populate a `crd.Controller` from **yaml** files on disk. The monitor will watch
+for changes to yaml files under a specified directory and will update a `crd.Controller` when appropriate.
+
+
+# Creating a Monitor
+To create a monitor, you just need to provide the `crd.Controller`, the directory to be monitored, and (optionally)
+the config types.
+
+```golang
+monitor := file.NewMonitor(
+    controller,      // The crd controller
+    fsroot,          // The root dir to monitor for yaml files
+    nil)             // Spported types (optional)
+```
+
+# Running a Monitor
+Once created, you simply run the monitor, providing a stop channel.
+
+```golang
+stop := make(chan struct{})
+...
+go monitor.Run(stop)
+```
+
+# Notes
+## Always use a Controller
+While the API supports any `model.ConfigStore`, it is recommneded to always use a `crd.Controller` so that other
+system components can be notified of changes via `controller.RegisterEventHandler()`.
+## Update Rate
+The timer used for polling the files is curretnly fixed at 1/2 second. This may be made configurable in the future
+if it's determined to be useful.
+## Supported Types
+If no config types are provided to `NewMonitor` (either `nil` or empty array), the default types assumed will be
+`model.IstioConfigTypes` (i.e. everything).
+## Example
+See `monitor_test.go` for a concrete example.
```

Comment:
Excellent. A meta level question: why can't we have the concrete example be part of the main code base? Like a default controller with file watcher? Not many are going to be implementing this on their own. i.e., move monitor+test.go's filewatcher into monitor.go ?

Reply:
I'm not sure how helpful this method would be.  Any users of this would just have to write 3 lines of code:

```golang
store := memory.Make(model.IstioConfigTypes)
controller = memory.NewController(store)
monitor = file.NewMonitor(controller, fsroot, nil)
```

I'm hesitant to add helper methods that will be unused at the time the PR is written. We can revisit when we have multiple uses of the file monitor.